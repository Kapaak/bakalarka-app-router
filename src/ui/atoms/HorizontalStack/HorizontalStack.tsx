import { HTMLAttributes, PropsWithChildren } from "react";

import { VariantProps, cva } from "class-variance-authority";

const horizontalStackVariant = cva("flex", {
  variants: {
    justify: {
      normal: ["justify-normal"],
      start: ["justify-start"],
      center: ["justify-center"],
      between: ["justify-center"],
      around: ["justify-center"],
      evenly: ["justify-evenly"],
    },
    align: {
      stretch: ["items-stretch"],
      start: ["items-start"],
      end: ["items-end"],
      baseline: ["items-baseline"],
      center: ["items-center"],
    },
  },
  defaultVariants: {
    justify: "normal",
    align: "stretch",
  },
});

export type HorizontalStackProps = HTMLAttributes<HTMLDivElement> &
  VariantProps<typeof horizontalStackVariant>;

export const HorizontalStack = ({
  justify,
  align,
  children,
  className,
}: PropsWithChildren<HorizontalStackProps>) => {
  return (
    <div className={horizontalStackVariant({ justify, align, className })}>
      {children}
    </div>
  );
};
