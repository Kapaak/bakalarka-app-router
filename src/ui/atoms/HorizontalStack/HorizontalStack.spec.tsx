import { render } from "@testing-library/react";

import { HorizontalStack } from "./HorizontalStack";

describe("HorizontalStack", () => {
  it("should render successfully", () => {
    const { baseElement } = render(<HorizontalStack />);

    expect(baseElement).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });
});
