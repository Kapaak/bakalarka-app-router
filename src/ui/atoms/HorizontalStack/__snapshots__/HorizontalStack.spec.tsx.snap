// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalStack should render successfully 1`] = `
<body>
  <div>
    <div
      class="flex justify-normal items-stretch"
    />
  </div>
</body>
`;
